pretyping/coercion.ml

```diff
@@ -522,38 +522,76 @@ let reapply_coercions_body sigma trace body =
     let body = reapply_coercions sigma trace body in
     start_app_body sigma body
 
+type hook = env -> evar_map -> flags:Evarconv.unify_flags -> constr ->
+  inferred:types -> expected:types -> (evar_map * constr) option
```

Comment:
```suggestion
type expected = Type of types | Sort | Product
type hook = env -> evar_map -> flags:Evarconv.unify_flags -> constr ->
  inferred:types -> expected:expected -> (evar_map * constr) option
```

Reply:
Thank you, I updated the rest of the code accordingly (with some small cleanup), updated the hook in coq-elpi and my small test still succeeds.